Windows/CommonDialog/MainWindow.xaml.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System.Text;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using System.IO;
+using Microsoft.Win32;
+using System.Diagnostics;
+using System.ComponentModel;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace CommonDialogs
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            InitializeComponent();
+            DataContext = new DataModel();
+        }
+
+        private void Button_OpenFileDialog(object sender, RoutedEventArgs e)
+        {
+            // Instantiate the open file dialog box.
+            // Setting the following properties are optional
+            OpenFileDialog openFileDialog = new OpenFileDialog()
+            {
+                Title = "Select any text file...",
+                InitialDirectory = @"C:\Windows",
+                Filter = "Text files (*.txt)|*.txt"
+            };
+
+            // Forces the preview pane to be shown irrespective of the system settings
+            //openFileDialog.ForcePreviewPane = true;
+
+            // Allows to select multiple files in the dialog
+            openFileDialog.Multiselect = true;
+
+
+            string text = "";
+
+            // If the user clicks OK button on the dialog, the ShowDialog() method returns true
+            // otherwise it returns false.
+            if (openFileDialog.ShowDialog() == true)
+            {
+                var streams = openFileDialog.OpenFiles();
+                foreach (var stream in streams)
+                {
+                    using (stream)
+                    {
+                        using (var reader = new System.IO.StreamReader(stream))
+                        {
+                            text += reader.ReadToEnd();
+                        }
+                        text += "\n\n--- END OF FILE ---\n";
```

Comment:
A better practice would be to use `StringBuilder` here. Also `\n` is not the new line separator on Windows, but if you switch to SB, `AppendLine()` will take care of that.